server/src/main/java/org/elasticsearch/action/search/SearchRequest.java

```diff
@@ -243,6 +256,17 @@ String getLocalClusterAlias() {
         return localClusterAlias;
     }
 
+    /**
+     * Returns the current time in milliseconds from the time epoch, to be used for the execution of this search request. A non-null
+     * value (expected to be greater or equal than 0) indicates that this search request is being executed as part of a locally reduced
+     * cross-cluster search request. The provided current time is used to ensure that the same value, determined by the CCS coordinating
+     * node, is used on all clusters involved in the execution of the search request.
+     */
+    @Nullable
+    Long getAbsoluteStartMillis() {
+        return absoluteStartMillis == DEFAULT_ABSOLUTE_START_MILLIS ? null : absoluteStartMillis;
```

Comment:
Since this is called only once it could be called getOrCreateAbsoluteStartMillis() and return `System.currentTimeMillis()` if the `absoluteStartMillis` is not set (equals to `-1) ?

Reply:
yes I will do that

Reply:
we could even use currentTimeMillis as a default in the constructor considering when the search request gets created on the coord node, if it wasn't for the transport client, in which case that would take the current time on the client instead of on the server.. Too bad :)